src/baseTrie.ts

```diff
@@ -787,41 +699,35 @@ export class Trie {
    * @method batch
    * @memberof Trie
    * @example
-   * var ops = [
+   * const ops = [
    *    { type: 'del', key: Buffer.from('father') }
    *  , { type: 'put', key: Buffer.from('name'), value: Buffer.from('Yuri Irsenovich Kim') }
    *  , { type: 'put', key: Buffer.from('dob'), value: Buffer.from('16 February 1941') }
    *  , { type: 'put', key: Buffer.from('spouse'), value: Buffer.from('Kim Young-sook') }
    *  , { type: 'put', key: Buffer.from('occupation'), value: Buffer.from('Clown') }
    * ]
-   * trie.batch(ops)
+   * await trie.batch(ops)
    * @param {Array} ops
-   * @param {Function} cb
+   * @returns {Promise}
    */
-  batch(ops: BatchDBOp[], cb: ErrorCallback) {
-    async.eachSeries(
-      ops,
-      (op: BatchDBOp, cb2: ErrorCallback) => {
-        if (op.type === 'put') {
-          if (!op.value) throw new Error('Invalid batch db operation')
-          this.put(op.key, op.value, cb2)
-        } else if (op.type === 'del') {
-          this.del(op.key, cb2)
-        } else {
-          cb2()
+  async batch(ops: BatchDBOp[]): Promise<void> {
+    for await (const op of ops) {
+      if (op.type === 'put') {
+        if (!op.value) {
+          throw new Error('Invalid batch db operation')
         }
-      },
-      cb,
-    )
+        await this.put(op.key, op.value)
+      } else if (op.type === 'del') {
+        await this.del(op.key)
+      }
+    }
   }
 
   /**
-   * Checks if a given root exists
+   * Checks if a given root exists.
    */
-  checkRoot(root: Buffer, cb: Function) {
-    root = ethUtil.toBuffer(root)
-    this._lookupNode(root, (e: Error, value: TrieNode) => {
-      cb(null, !!value)
-    })
+  async checkRoot(root: Buffer): Promise<boolean> {
+    const value = await this._lookupNode(root)
+    return !!value
```

Comment:
Phew, nice. 😀

Apart from `walkTrie` needing a second look the rest from `baseTrie.js` looks good.